sys/net/gnrc/application_layer/dhcpv6/client.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include "log.h"
+#include "net/arp.h"
+#include "net/dhcpv6.h"
+#include "net/gnrc/ipv6/nib/pl.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/rpl.h"
+#include "net/sock.h"
+#include "timex.h"
+
+#include "net/dhcpv6/client.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+unsigned dhcpv6_client_get_duid_l2(unsigned iface, dhcpv6_duid_l2_t *duid)
+{
+    gnrc_netif_t *netif;
+    uint8_t *l2addr = ((uint8_t *)(duid)) + sizeof(dhcpv6_duid_l2_t);
+    int res;
+
+    duid->type = byteorder_htons(DHCPV6_DUID_TYPE_L2);
+    /* TODO make GNRC-independent */
+    if (iface == SOCK_ADDR_ANY_NETIF) {
+        netif = gnrc_netif_iter(NULL);
```

Comment:
So it will always use only the first interface?

Reply:
Yes, this is just a prototype...

Reply:
Also no: The first interface is only selected if no interface is provided. This is the current config in the stack-independent part of the DHCPv6 client, however, if iface is set to anything but `SOCK_ADDR_ANY_NETIF` that interface is selected. See below.